File: src/engine/primitives/image.ts
/**
 * image.ts
 *
 * Slot path constants for the image type are all this file holds. No logic
 * joins them.
 *
 * The image primitive is data plus one renderer arm. It stores the picture as
 * a data URL in the document.
 *
 * Engine-layer code: pure logic with no DOM, window or canvas access, so the
 * tests run headless and the file can move to Rust later.
 */
export const IMAGE_WIDTH_PATH: readonly string[] = ["width"];
export const IMAGE_HEIGHT_PATH: readonly string[] = ["height"];

export const IMAGE_OPACITY_PATH: readonly string[] = ["opacity"];

export const IMAGE_PRESERVE_ASPECT_PATH: readonly string[] = ["preserveAspect"];

export const IMAGE_PICTURE_ASPECT_PATH: readonly string[] = ["pictureAspect"];

export const IMAGE_SOURCE_PATH: readonly string[] = ["source"];

File: crates/beheader-engine/src/primitives/image.rs
//! Slot path constants for the image type, and nothing else. There is no logic
//! in this file.
//!
//! An image is data plus one arm in the renderer. The picture itself is stored
//! as a data URL inside the document, so a saved file carries its images with
//! it.
//!
//! The port of `src/engine/primitives/image.ts`.

fn path(segments: &[&str]) -> Vec<String> {
    segments.iter().map(|part| (*part).to_string()).collect()
}

pub fn image_width_path() -> Vec<String> {
    path(&["width"])
}

pub fn image_height_path() -> Vec<String> {
    path(&["height"])
}

pub fn image_opacity_path() -> Vec<String> {
    path(&["opacity"])
}

pub fn image_preserve_aspect_path() -> Vec<String> {
    path(&["preserveAspect"])
}

pub fn image_picture_aspect_path() -> Vec<String> {
    path(&["pictureAspect"])
}

pub fn image_source_path() -> Vec<String> {
    path(&["source"])
}
